pyproject.toml

```diff
@@ -12,18 +12,37 @@ dependencies = [
     "boto3",
     "pandas",
     "levenshtein",
-    "pathlib==1.0.1",
-    "python-dotenv==1.0.1",
-    "scikit-learn==1.4.0",
-    "click==8.1.7",
-    "PyYAML==6.0.1",
-    "langdetect==1.0.9",
+    "pathlib",
+    "python-dotenv",
+    "scikit-learn>=1.4.0",
+    "click>=8.0.0",
+    "PyYAML>=6.0.1",
+    "langdetect>=1.0.9",
+    "regex",
+    "amazon-textract-textractor",
     "backoff",
-    "PyMuPDF==1.23.26",
+    "PyMuPDF>=1.23.26",
     "opencv-python-headless",
-    "quads==1.1.0",
+    "quads>=1.1.0",
 ]
 
+[project.optional-dependencies]
```

Comment:
For the optional dependencies, we don't need to be that strict. I.e. we don't need to weaken the version as this is very much related to development and not for production purposes.